OpenRA.Mods.Common/Traits/Render/ProductionBar.cs

```diff
@@ -16,14 +16,33 @@
 namespace OpenRA.Mods.Common.Traits.Render
 {
 	[Desc("Visualizes the remaining build time of actor produced here.")]
-	class ProductionBarInfo : ConditionalTraitInfo, Requires<ProductionInfo>
+	class ProductionBarInfo : ConditionalTraitInfo, Requires<ProductionInfo>, IRulesetLoaded
 	{
 		[FieldLoader.Require]
 		[Desc("Production queue type, for actors with multiple queues.")]
 		public readonly string ProductionType = null;
 
 		public readonly Color Color = Color.SkyBlue;
 
+		public override void RulesetLoaded(Ruleset rules, ActorInfo ai)
+		{
+			ProductionQueueInfo queue;
```

Comment:
Actually, thinking about it: It would be better to just do `var queue = ...` below.

Reply:
Okay.